third_party/move/move-compiler-v2/src/file_format_generator/function_generator.rs

```diff
@@ -512,7 +512,16 @@ impl<'a> FunctionGenerator<'a> {
     ) {
         let fun_ctx = ctx.fun_ctx;
         self.abstract_push_args(ctx, source);
-        if inst.is_empty() {
+        if let Some(opcode) = ctx.fun_ctx.module.get_well_known_function_code(
```

Comment:
I note that V1 code checks (in fake_natives.rs) for this check for the existence of an attribute (AttributeName_::Known(KnownAttribute::Native(NativeAttribute::BytecodeInstruction))) before trying to match a "fake native"/"well-known function".  I might agree with their comment about developer sanity here:
```
//! This module verifies the usage of the "fake native" functions. These functions are declared
//! as 'native`, but do not appear in the compiled module. For developer sanity, they must be marked
//! with the `FAKE_NATIVE_ATTR`
```
If someday our descendants decide to add a new well-known function they might have an easier time if they get a warning from the compiler that it sees the #[bytecode_instruction] annotation on the function definition but doesn't know how to translate it.

Reply:
I'm not sure whether there is any value to this attribute, and I remember the author wasn't sure either. The attribute would have to at least also denote the bytecode which is going to be used. In fact, I think those functions should not appear in the first place in vector, a refactoring we may want to do at a later point. (Specification builtin functions do not have source code.) They have already introduced quite some confusion including security incidents.